packages/zapp/console/src/components/Launch/LaunchForm/inputHelpers/map.ts

```diff
@@ -0,0 +1,80 @@
+import { stringifyValue } from 'common/utils';
+import { Core } from 'flyteidl';
+import { InputTypeDefinition, InputValue } from '../types';
+import { getHelperForInput } from './getHelperForInput';
+import { ConverterInput, InputHelper, InputValidatorParams } from './types';
+import t from '../../../common/strings';
+
+function fromLiteral(literal: Core.ILiteral, { subtype }: InputTypeDefinition): InputValue {
+  if (!subtype) {
+    throw new Error(t('missingMapSubType'));
+  }
+  if (!literal.map) {
+    throw new Error('Map literal missing `map` property');
+  }
+  if (!literal.map.literals) {
+    throw new Error('Map literal missing `map.literals` property');
+  }
+  if (typeof literal.map.literals !== 'object') {
+    throw new Error('Map literal is not an object');
+  }
+  if (!Object.keys(literal.map.literals).length) {
+    throw new Error('Map literal object is empty');
+  }
+
+  const key = Object.keys(literal.map.literals)[0];
+  const childLiteral = literal.map.literals[key];
+  const helper = getHelperForInput(subtype.type);
+
+  return stringifyValue({ [key]: helper.fromLiteral(childLiteral, subtype) });
+}
+
+function toLiteral({ value, typeDefinition: { subtype } }: ConverterInput): Core.ILiteral {
+  if (!subtype) {
+    throw new Error(t('missingMapSubType'));
+  }
+  const obj = JSON.parse(value.toString());
+  const key = Object.keys(obj)?.[0];
+
+  const helper = getHelperForInput(subtype.type);
+
+  return {
+    map: { literals: { [key]: helper.toLiteral({ value: obj[key], typeDefinition: subtype }) } },
+  };
+}
+
+function validate({ value, typeDefinition: { subtype } }: InputValidatorParams) {
+  if (!subtype) {
+    throw new Error(t('missingMapSubType'));
+  }
+  if (typeof value !== 'string') {
+    throw new Error('Value is not a string');
+  }
+  if (!value.toString().length) {
+    throw new Error('Value is required');
```

Comment:
Can you please add all strings mentioned in `Error(xxx)` in this file to string file.

Reply:
fixed